### PR TITLE
ci(*): add ci workflow

### PR DESCRIPTION
**Pull request by manudeli**
CI should check our package is valid for users
In my opinion, We should improve this procedure gradually

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  quality:
+    name: Check quality
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        command: ['ci:attw', 'ci:publint', 'ci:sherif', 'ci:type', 'test', 'build']
```
Comment (manudeli) [inline .github/workflows/ci.yml:18]: This change make us check automatically in Pull Request
1. attw: https://github.com/arethetypeswrong/arethetypeswrong.github.io
2. publint: https://github.com/bluwy/publint
3. sherif: https://github.com/QuiiBz/sherif
4. tsc: TypeScript check
5. build: Build All

File: configs/tsup/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@suspensive/tsconfig/base.json",
+  "extends": "@coauthors/tsconfig/base.json",
```
Comment (manudeli) [inline configs/tsup/tsconfig.json:2]: This ci:type make me check this automatically

File: packages/core/package.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
     },
     "./package.json": "./package.json"
   },
-  "main": "src/index.js",
+  "main": "dist/index.js",
```
Comment (manudeli) [inline packages/core/package.json:33]: This ci:publint make me check this automatically

File: packages/core/src/coauthor.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { githubUser } from "./utils/githubUser";
 export const coauthor = (author: z.infer<typeof authorSchema>) =>
   githubUser(authorSchema.parse(author).user).then(
     (githubUser) =>
-      `Co-authored-by: ${author.name ?? githubUser.name} <${githubUser.id}+${
-        author.user
-      }@users.noreply.github.com>` as const,
+      `Co-authored-by: ${author.name ?? githubUser.name ?? author.user} <${
+        githubUser.id
+      }+${author.user}@users.noreply.github.com>` as const,
```
Comment (manudeli) [inline packages/core/src/coauthor.ts:10]: This ci:test make me improve this automatically

**Review by manudeli — COMMENTED** [review]
I left my intentions

File: packages/core/src/coauthor.ts (hunk 4 above)
Comment (2-NOW) [reply packages/core/src/coauthor.ts:10]: If there is no github display name, it appears that the implementation uses github user id instead. great !

**Review by 2-NOW — APPROVED** [review]

**Pull request merged**